### PR TITLE
Sync with upstream and update .gitignore

### PR DESCRIPTION
**Pull request by NereusWB922**
Upstream changes
- Add puml files for tutorial class diagrams

Update .gitignore file
- Include bin folder

### CODE REVIEW

**Comment by mingyuanc** [comment]
Looks good!

**Review by mingyuanc — APPROVED** [review]
Looks good!

File: docs/diagrams/add-remark/ParserClass.puml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+@startuml
```
Comment (Nid21cs) [inline docs/diagrams/add-remark/ParserClass.puml:1]: Nice follows coding standards!

**Comment by Nid21cs** [comment]
Nice

**Pull request merged**